presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java

```diff
@@ -216,8 +210,9 @@ public void queryCompletedEvent(QueryInfo queryInfo)
                     plan = Optional.of(textDistributedPlan(
                             queryInfo.getOutputStage().get(),
                             functionRegistry,
-                            statsCalculator,
-                            costCalculator,
+                            // Not possible to recompute stats and costs, since transaction is already completed at this moment
```

Comment:
"recompute" or "compute"?

Reply:
Sorry, missed that comment. Let's leave it as "recompute". Since in theory up to this moment statistics could've been computed at least once when doing cost based optimizations. 

We are going to remove this soon, after we fix the issue. As we need the stats in the final plan.